subprojects/core-api/src/main/java/org/gradle/api/tasks/util/internal/PatternSpecFactory.java

```diff
@@ -113,7 +114,12 @@ public synchronized void setDefaultExcludesFromSettings(String[] excludesFromSet
         }
     }
 
-    private void updateDefaultExcludeSpecCache(String[] defaultExcludes) {
+    @VisibleForTesting
+    public synchronized List<String> getDefaultExcludesFromSettings() {
```

Comment:
I don't think you need to expose this method. You can call `DirectoryScanner.getDefaultExcludes()` from the test.

Reply:
`DirectoryScanner.getDefaultExcludes()` would return the empty list in this case, where I need the "previous value".

Reply:
You can call `DirectoryScanner.reset...` to return to the default value.

Reply:
I'm not sure what you mean: this method is used in a test to return the list of default excludes configured in settings. This is only used to print the list. I don't use them to reset :thinking:

Reply:
In your test, you don't need this method. You can remember the previous state from `DirectoryScanner.getDefaultExcludes()` to check the exception message. I would prefer removing this method from `PatternSpecFactory`.

Reply:
ah now I see what you mean :)